src/components/Switch/CustomSwitch.tsx

```diff
@@ -0,0 +1,180 @@
+import { forwardRef, memo, useCallback, useEffect, useState, useMemo } from 'react';
+import { Animated, Easing, NativeModules, Platform, StyleSheet, SwitchProps } from 'react-native';
+import { useComponentStyles, useMolecules } from '../../hooks';
+import setColor from 'color';
+
+const version = NativeModules.PlatformConstants
+    ? NativeModules.PlatformConstants.reactNativeVersion
+    : undefined;
+
+type ToggleProps = SwitchProps & {
+    label?: string;
+    labelStyle?: object;
+    onIcon?: string;
+    offIcon?: string;
+};
+
+const Switch = ({
+    trackColor = { false: 'colors.onSurface', true: 'colors.primary' },
+    thumbColor = 'green',
+    label = '',
+    onValueChange,
+    disabled,
+    value,
+    labelStyle = {},
+    onIcon,
+    offIcon,
+    style,
+}: ToggleProps) => {
+    const { Icon, TouchableRipple, View, Text } = useMolecules();
+    const componentStyles = useComponentStyles(
+        'NewSwitch',
```

Comment:
why do we have switch in this branch?

Reply:
I pushed it by mistake, will delete it from this branch.